src/content/peerconnection/video-analyzer/js/main.js

```diff
@@ -0,0 +1,232 @@
+    onCreateSessionDescriptionError(e);
+  }
+}
+
+function onSetLocalSuccess(pc) {
+  console.log(`${getName(pc)} setLocalDescription complete`);
+}
+
+function onSetRemoteSuccess(pc) {
+  console.log(`${getName(pc)} setRemoteDescription complete`);
+}
+
+function onSetSessionDescriptionError(error) {
+  console.log(`Failed to set session description: ${error.toString()}`);
+}
+
+function gotRemoteTrack(e) {
+  console.log('pc2 received remote stream');
+  const frameStreams = e.receiver.createEncodedVideoStreams();
+  frameStreams.readableStream.pipeThrough(new TransformStream({
+    transform: videoAnalyzer
+  }))
+      .pipeTo(frameStreams.writableStream);
+  remoteVideo.srcObject = e.streams[0];
+}
+
+async function onCreateAnswerSuccess(desc) {
+  console.log(`Answer from pc2:\n${desc.sdp}`);
+  console.log('pc2 setLocalDescription start');
+  try {
+    await pc2.setLocalDescription(desc);
+    onSetLocalSuccess(pc2);
+  } catch (e) {
+    onSetSessionDescriptionError(e);
+  }
+  console.log('pc1 setRemoteDescription start');
+  try {
+    await pc1.setRemoteDescription(desc);
+    onSetRemoteSuccess(pc1);
+  } catch (e) {
+    onSetSessionDescriptionError(e);
+  }
+}
+
+async function onIceCandidate(pc, event) {
+  try {
+    await (getOtherPc(pc).addIceCandidate(event.candidate));
+    onAddIceCandidateSuccess(pc);
+  } catch (e) {
+    onAddIceCandidateError(pc, e);
+  }
+  console.log(`${getName(pc)} ICE candidate:\n${event.candidate ? event.candidate.candidate : '(null)'}`);
+}
+
+function onAddIceCandidateSuccess(pc) {
+  console.log(`${getName(pc)} addIceCandidate success`);
+}
+
+function onAddIceCandidateError(pc, error) {
+  console.log(`${getName(pc)} failed to add ICE Candidate: ${error.toString()}`);
+}
+
+function onIceStateChange(pc, event) {
+  if (pc) {
+    console.log(`${getName(pc)} ICE state: ${pc.iceConnectionState}`);
+    console.log('ICE state change event: ', event);
+  }
+}
+
+function hangup() {
+  console.log('Ending call');
+  pc1.close();
+  pc2.close();
+  pc1 = null;
+  pc2 = null;
+  hangupButton.disabled = true;
+  callButton.disabled = false;
+}
+
+const keyFrameCountDisplay = document.querySelector('#keyframe-count');
+const keyFrameSizeDisplay = document.querySelector('#keyframe-size');
+const interFrameCountDisplay = document.querySelector('#interframe-count');
+const interFrameSizeDisplay = document.querySelector('#interframe-size');
+let keyFrameCount = 0;
+let interFrameCount = 0;
+
+function videoAnalyzer(chunk, controller) {
+  const view = new DataView(chunk.data);
+  // The lowest value bit in the first byte is the keyframe indicator.
+  const keyframeBit = view.getUint8(0) & 0x01;
+  // console.log(view.getUint8(0).toString(16));
+  if (keyframeBit === 0) {
+    keyFrameCount++;
+    keyFrameCountDisplay.innerText = keyFrameCount;
+    keyFrameSizeDisplay.innerText = chunk.data.byteLength;
+  } else {
+    interFrameCount++;
+    interFrameCountDisplay.innerText = interFrameCount;
```

Comment:
this is going to update *very* rapidly. Increment and update the display once a second?

Reply:
setTimeout()? Once a second is a bit too slow, it feels like. Trying every half second.

Reply:
`requestAnimationFrame` is a better choice?

Reply:
@leader22 requestAnimationFrame will probably fire faster than once-per-frame, so if the point is to make less flicker, setTimeout() seems good.